boot-environ/src/environ/boot.clj

```diff
@@ -1,10 +1,45 @@
 (ns environ.boot
   {:boot/export-tasks true}
   (:require [boot.core :as core]
-            [environ.core :as environ]))
+            [environ.core :as environ]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]))
 
-(core/deftask environ [e env FOO=BAR {kw str} "The environment map"]
-  (fn [next-task]
-    (fn [fileset]
-      (with-redefs [environ/env (merge environ/env env)]
-        (next-task fileset)))))
+(def ^:const env-file ".boot-env")
+
+(defn- read-boot-env [fileset]
+  (if-let [t (core/tmp-get fileset env-file)]
+    (-> t
+        core/tmp-file
+        slurp
+        edn/read-string)
+    {}))
+
+(defn- write-boot-env [env tmp]
+  (let [s (prn-str env)
+        f (io/file tmp env-file)]
+    (spit f s)))
+
+(defn- update-boot-env
+  "Merges the given environment (env) with the environment in the fileset's
+   .boot-env file; writes the result to the temporary directory tmp."
+  [fileset tmp env]
+  (-> fileset
+      read-boot-env
+      (merge env)
+      (write-boot-env tmp)))
+
+(core/deftask environ
+  "Adds key-value pairs to the environment picked up by environ."
+  [e env       FOO=BAR {kw str} "The environment map"
+   b boot-env          bool     "Create/update .boot-env file in fileset"]
```

Comment:
Is there ever a reason we *wouldn't* want a `.boot-env` file? It seems to me that we could just do this by default and forgo the additional option.